lib/Driver/Driver.cpp

```diff
@@ -703,6 +703,17 @@ Driver::buildCompilation(const ToolChain &TC,
   OI.CompilerMode = computeCompilerMode(*TranslatedArgList, Inputs, BatchMode);
   buildOutputInfo(TC, *TranslatedArgList, BatchMode, Inputs, OI);
 
+  // Note: Batch mode handling of serialized diagnostics requires that all
+  // batches get to run. (This behavior could be limited to only when serialized
+  // diagnostics are being emitted, but this seems more consistent and less
+  // surprising for users.)
+  // FIXME: We don't really need (or want) a full ContinueBuildingAfterErrors.
+  // If we fail to precompile a bridging header, for example, there's no need
+  // to go on to compilation of source files, and if compilation of source files
+  // fails, we shouldn't try to link. Instead, we'd want to let all jobs finish
+  // but not schedule any new ones.
+  ContinueBuildingAfterErrors |= BatchMode;
```

Comment:
`ContinueBuildingAfterErrors` isn't used before this, so how about moving the earlier code that reads the argument and sets it down to here? That way it could be a `const bool` instead of just a `bool`.

Reply:
Also, consider adding an assertion `ContinueBuildingAfterErrors || !BatchMode` around line 531 in Compilation.cpp
where it says ` return Comp.ContinueBuildingAfterErrors ? ... : ...`